tasks/smoke.yml

```diff
@@ -9,6 +9,7 @@ inputs:
 - name: concourse
 - name: ci
 - name: endpoint-info
+- name: version
```

Comment:
`smoke.yml` gets consumed in many places:
* `bin-smoke`, `bin-smoke-kernel`, `bin-smoke-containerd`, `bosh-smoke` in `concourse.yml` - this pipeline is quite intentional about _not_ having a semver resource as it purely produces dev builds. How we gonna do a version comparison there?
* `bin-smoke`, `bin-smoke-containerd`, `bosh-smoke` in `release-v6.yml` - I think these all have a semver resource called `version` in their build context so we're probably safe here.
* various and sundry spots in branch/drills/legacy pipelines

Reply:
Whoops, this change should've been in `terraform-smoke.yml` instead.

What do you think about making terraform dumb and only capable of emitting/omitting stuff based on flags. This way we can pass the flags directly to the task similar to how we  https://github.com/concourse/ci/blob/f8e639f72da8961327c83665843ef1cd03e226fa/pipelines/concourse.yml#L753

This would also allow us to move the version comparison logic into the bash script (which can set/unset env vars as needed)

- `concourse.yml`: uses default
- `release.yml` (I believe this is legacy release pipeline): add a `load_var` step to extract the version from semver resource and pass it in
- `release-v6.yml`: same as above
- `branch.yml`:  add a pipeline variable (`compatibility_version` or something) that has to be passed in
- `drill.yml`: not used by the reconfigure pipeline, but can use the same behaviour as above